### PR TITLE
x86: add multiboot2 support

### PR DESCRIPTION
**Pull request by alex-ab**
Issue #67

### CODE REVIEW

File: src/plat/pc99/machine/acpi.c [hunk 1]
```diff
@@ -244,20 +215,34 @@ acpi_table_init(void* entry, enum acpi_type table_type)
...
 
-    if (acpi_rsdp == NULL) {
+    printf("ACPI: RSDP vaddr=%p\n", acpi_rsdp);
+
+    if (!acpi_rsdp || acpi_calc_checksum((char const *)acpi_rsdp, 20) != 0) {
         printf("BIOS: No ACPI support detected\n");
         return NULL;
     }
-    printf("ACPI: RSDP paddr=%p\n", acpi_rsdp);
-    acpi_rsdp = acpi_table_init(acpi_rsdp, ACPI_RSDP);
```
Comment (AdrianDanis) [inline src/plat/pc99/machine/acpi.c]: I'm trying to understand this change. It seems like `acpi_search_rsdp` is returning the mapped rsdp address, but that only gets called from a multiboot1 boot. When using the value from multiboot2 as far as I can tell `acpi_table_init` does not get called, and we are just using the raw pointer. It isn't clear to me why that is guaranteed to be mapped in?

**Comment by AdrianDanis** [comment]
I have some minor nitpicks and changes I would like to make to this, both in the code as well as splitting up into multiple commits with extended messages. Given you have already done the bulk of the implementation thus far I am happy to do these and leave your name on the commits if you do not have time, otherwise I can do more detailed comments in this PR for you.

File: src/plat/pc99/machine/acpi.c (hunk 1 above)
Comment (alex-ab) [reply src/plat/pc99/machine/acpi.c]: > I'm trying to understand this change. It seems like acpi_search_rsdp is returning the mapped rsdp address, but that only gets called from a multiboot1 boot.

Yes.

> When using the value from multiboot2 as far as I can tell acpi_table_init does not get called, and we are just using the raw pointer. It isn't clear to me why that is guaranteed to be mapped in?

Multiboot 2 provides a  copy of the ACPI RSDP data, directly embedded within the already mapped Multiboot 2 structure. So, the pointer points into the dense packed MBI2 structure. (This seems in general the main difference to MBI1, where all parts of MBI information may lie in arbitrary memory locations connected by pointers.)

**Comment by alex-ab** [comment]
> Given you have already done the bulk of the implementation thus far I am happy to do these and leave your name on the commits if you do not have time.

I'm fine with that you take over, thank you.

File: src/plat/pc99/machine/acpi.c (hunk 1 above)
Comment (AdrianDanis) [reply src/plat/pc99/machine/acpi.c]: Ahh I didn't pick this up from my quick skim of the multiboot2 spec. Thanks.

**Comment by AdrianDanis** [comment]
Closing now that this has been added in 4241b2df6fb7cd182277d984324c0b3186f4588c

**Pull request closed without merging**